spot_micro_joy/scripts/spotMicroJoystickMove.py

```diff
@@ -0,0 +1,150 @@
+        self._stand_event_cmd_msg.data = True
+
+        self._idle_event_cmd_msg = Bool()
+        self._idle_event_cmd_msg.data = True
+
+        rospy.loginfo("Setting Up the Spot Micro Joystick Control Node...")
+
+        # Set up and title the ros node for this code
+        rospy.init_node('spot_micro_joystick_control')
+
+        # Create publishers for commanding velocity, angle, and robot states
+        self._ros_pub_angle_cmd = rospy.Publisher('/angle_cmd', Vector3, queue_size=1)
+        self._ros_pub_vel_cmd = rospy.Publisher('/cmd_vel', Twist, queue_size=1)
+        self._ros_pub_walk_cmd = rospy.Publisher('/walk_cmd', Bool, queue_size=1)
+        self._ros_pub_stand_cmd = rospy.Publisher('/stand_cmd', Bool, queue_size=1)
+        self._ros_pub_idle_cmd = rospy.Publisher('/idle_cmd', Bool, queue_size=1)
+
+        rospy.loginfo("Joystick control node publishers corrrectly initialized")
+
+    def reset_all_motion_commands_to_zero(self):
+        '''Reset body motion cmd states to zero and publish zero value body motion commands'''
+
+        self._vel_cmd_msg.linear.x = 0
+        self._vel_cmd_msg.linear.y = 0
+        self._vel_cmd_msg.linear.z = 0
+        self._vel_cmd_msg.angular.x = 0
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._ros_pub_vel_cmd.publish(self._vel_cmd_msg)
+
+    def reset_all_angle_commands_to_zero(self):
+        '''Reset angle cmd states to zero and publish them'''
+
+        self._angle_cmd_msg.x = 0
+        self._angle_cmd_msg.y = 0
+        self._angle_cmd_msg.z = 0
+
+        self._ros_pub_angle_cmd.publish(self._angle_cmd_msg)
+
+    def on_joy(self, msg):
+        if msg.buttons[self.BUTTON_GREEN] == 1:
+            self._ros_pub_idle_cmd.publish(self._idle_event_cmd_msg)
+            rospy.loginfo('Idle command issued from joystick.')
+            self.mode = self.MODE_IDLE
+
+        if msg.buttons[self.BUTTON_YELLOW] == 1:
+            self._ros_pub_stand_cmd.publish(self._stand_event_cmd_msg)
+            rospy.loginfo('Stand command issued from joystick.')
+            self.mode = self.MODE_STAND
+
+        if msg.buttons[self.BUTTON_BLUE] == 1:
+            self.reset_all_angle_commands_to_zero()
+            rospy.loginfo('Entering joystick angle command mode.')
+            self.mode = self.MODE_ANGLE
+
+        if msg.buttons[self.BUTTON_RED] == 1:
+            self.reset_all_angle_commands_to_zero()
+            self._ros_pub_walk_cmd.publish(self._walk_event_cmd_msg)
+            rospy.loginfo('Entering joystick walk command mode.')
+            self.mode = self.MODE_WALK
+
+        if self.mode == self.MODE_ANGLE:
+            print('Cmd Values: phi: %1.3f deg, theta: %1.3f deg, psi: %1.3f deg ' \
+                  % (
+                  self._angle_cmd_msg.x * 180 / pi, self._angle_cmd_msg.y * 180 / pi, self._angle_cmd_msg.z * 180 / pi))
+
+            self._angle_cmd_msg.x = pi / 180 * msg.axes[self.AXES_ROLL] * 45 * -1
```

Comment:
What does the value 45 here represent?

Reply:
Joystick axes values are always between -1 and 1 which acts like a factor. so these are the max values to get by the analog sticks. I will extract them so the can get a name .. and are easy to change :)

Reply:
Yes that makes sense. Breaking them out either to a ROS parameter, or just a top level configurable variable would be a better design.

Reply:
extracted to top-level variables. this should be for each model the same, so i think it is not necessary to put it into the ROS configuration.